dbt_subprojects/hourly_spellbook/models/_sector/perpetual/projects/fxdx/base/fxdx_base_perpetual_trades.sql

```diff
@@ -0,0 +1,150 @@
+        collateralToken,
+        indexToken,
+        isLong,
+        CAST(realisedPnl AS DOUBLE)/1e30 AS pnl_usd,
+        CAST(collateral AS DOUBLE)/1e30 AS collateral_value,
+        CAST(reserveAmount AS DOUBLE)/1e30 AS reserve_amount
+    FROM {{ source('fxdx_base', 'Vault_evt_LiquidatePosition') }}
+    WHERE evt_block_time >= DATE '{{ project_start_date }}'
+)
+
+SELECT 
+    'base' AS blockchain,
+    'fxdx' AS project,
+    '1' AS version,
+    'fxdx' AS frontend,
+    CAST(DATE_TRUNC('day', pe.block_time) AS DATE) AS block_date,
+    CAST(DATE_TRUNC('month', pe.block_time) AS DATE) AS block_month,
+    pe.block_time,
+    CASE 
+        WHEN LOWER(CAST(pe.indexToken AS VARCHAR)) = '0x2ae3f1ec7f1f5012cfeab0185bfc7aa3cf0dec22' THEN 'cbETH'
+        WHEN LOWER(CAST(pe.indexToken AS VARCHAR)) = '0xd6c5469a7cc587e1e89a841fb7c102ff1370c05f' THEN 'WETH'
+        WHEN LOWER(CAST(pe.indexToken AS VARCHAR)) = '0xd9aaec86b65d86f6a7b5b1b0c42ffa531710b6ca' THEN 'USDbc'
+        ELSE CAST(pe.indexToken AS VARCHAR)
+    END AS virtual_asset,
+    CASE 
+        WHEN LOWER(CAST(pe.collateralToken AS VARCHAR)) = '0x2ae3f1ec7f1f5012cfeab0185bfc7aa3cf0dec22' THEN 'cbETH'
+        WHEN LOWER(CAST(pe.collateralToken AS VARCHAR)) = '0xd6c5469a7cc587e1e89a841fb7c102ff1370c05f' THEN 'WETH'
+        WHEN LOWER(CAST(pe.collateralToken AS VARCHAR)) = '0xd9aaec86b65d86f6a7b5b1b0c42ffa531710b6ca' THEN 'USDbc'
+        ELSE CAST(pe.collateralToken AS VARCHAR)
+    END AS underlying_asset,
+    CASE 
+        WHEN LOWER(CAST(pe.collateralToken AS VARCHAR)) = '0x2ae3f1ec7f1f5012cfeab0185bfc7aa3cf0dec22' THEN 'cbETH'
+        WHEN LOWER(CAST(pe.collateralToken AS VARCHAR)) = '0xd6c5469a7cc587e1e89a841fb7c102ff1370c05f' THEN 'WETH'
+        WHEN LOWER(CAST(pe.collateralToken AS VARCHAR)) = '0xd9aaec86b65d86f6a7b5b1b0c42ffa531710b6ca' THEN 'USDbc'
+        ELSE CAST(pe.collateralToken AS VARCHAR)
+    END AS market,
+    -- pe.position_id AS market,
+    pe.market_address,
+    pe.volume_usd,
+    pe.fee_usd,
+    pe.margin_usd,
+    CASE 
+        WHEN pe.trade_action = 'increase' AND pe.isLong = true THEN 'open_long'
+        WHEN pe.trade_action = 'increase' AND pe.isLong = false THEN 'open_short'
+        WHEN pe.trade_action = 'decrease' AND pe.isLong = true THEN 'close_long'
+        WHEN pe.trade_action = 'decrease' AND pe.isLong = false THEN 'close_short'
+        WHEN pe.trade_action = 'liquidate' THEN 'liquidation'
+    END AS trade,
+    pe.trader,
+    CASE 
+        WHEN pe.isLong = true THEN 'long'
+        WHEN pe.isLong = false THEN 'short'
+    END AS trade_type,
+    pe.price,
+    CAST(NULL AS UINT256) AS volume_raw,
+    pe.tx_hash,
+    pe.tx_to,
+    pe.tx_from,
+    pe.evt_index,
+    pe.pnl_usd AS pnl,
+    -- Additional FXDX specific fields
+    pe.collateral_value,
+    pe.reserve_amount
+FROM perp_events pe
+{% if is_incremental() %}
+    WHERE pe.block_time > (SELECT MAX(block_time) FROM {{ this }})
+{% endif %}
```

Comment:
please move incremental filters up to CTE as if/then statements with project start date filter. also use the standard incremental predicate macro rather than max value from target.